mozilla_schema_generator/glean_ping.py

```diff
@@ -13,20 +13,57 @@
 class GleanPing(GenericPing):
 
     schema_url = "https://raw.githubusercontent.com/mozilla-services/mozilla-pipeline-schemas/dev/schemas/glean/baseline/baseline.1.schema.json" # noqa E501
-    probes_url = "https://probeinfo.telemetry.mozilla.org/glean/{}/metrics"
+    probes_url_template = "https://probeinfo.telemetry.mozilla.org/glean/{}/metrics"
     repos_url = "https://probeinfo.telemetry.mozilla.org/glean/repositories"
 
-    default_probes_url = probes_url.format("glean")
+    default_dependencies = ['glean']
     default_pings = {"baseline", "events", "metrics"}
     ignore_pings = {"default", "glean_ping_info", "glean_client_info"}
 
     def __init__(self, repo):  # TODO: Make env-url optional
-        super().__init__(self.schema_url, self.schema_url, self.probes_url.format(repo))
+        self.repo = repo
+        super().__init__(
+            self.schema_url,
+            self.schema_url,
+            self.probes_url_template.format(repo)
+        )
+
+    def get_dependencies(self):
+        # Get all of the library dependencies for the application that
+        # are also known about in the repositories file.
+
+        # The dependencies are specified using library names, but we need to
+        # map those back to the name of the repository in the repository file.
+
+        dependencies = self._get_json(self.dependencies_url.format(self.repo))
+        dependency_library_names = list(dependencies.keys())
+
+        repos = self.get_repos()
+        repos_by_dependency_name = {}
+        for repo in repos:
+            for library_name in repo.get('library_names', []):
+                repos_by_dependency_name[library_name] = repo
+
+        dependencies = []
+        for name in dependency_library_names:
+            if name in repos_by_dependency_name:
+                dependencies.append(repos_by_dependency_name[name])
+
+        if not len(dependencies):
```

Comment:
Will this ever happen? Shouldn't at least glean be a dependency?

Reply:
If the application doesn't have any dependencies stored (such as when the dependencies artifact isn't being generated, as with the reference browser).  This is a test basically to make this fall back to having only glean as a dependency in that case.

Reply:
If that's the case, won't it return early above (line 46?)

Reply:
That's a good point.  It probably can't (assuming the `repositories.yaml` file has everything it needs and the dependencies artifact is correct).  I'll take it out.